apis/abortion_policy_api.py

```diff
@@ -50,58 +56,111 @@ def get_state_data():
+    keys_and_defaults = {}
+
+    # set default types of each state policy
+    for _, state_info in state_policies.items():
+        for k, v in state_info.items():
+            if k not in keys_and_defaults.keys():
+                key = re.sub(pattern, "", str(type(v))).split()[-1]
+                keys_and_defaults[k] = TYPE_DEFAULTS[key]
+
+    # fill in missing data for states currently in the dataset
+    for _, state_info in state_policies.items():
+        for k, v in keys_and_defaults.items():
+            if k not in state_info.keys():
+                state_info[k] = v
+
+    # add non-present states with default values
+    non_present_states = list(set(states) - set(state_policies.keys()))
+    for state in non_present_states:
+        state_policies[state] = keys_and_defaults
+
+    # sort dataset by state name
+    # code adapted from: https://www.geeksforgeeks.org/python-sort-a-dictionary/
+    state_policies = {
+        key: val for key, val in sorted(state_policies.items(), key=lambda ele: ele[0])
+    }
 
-    # sort states data
-    g_sort = sorted(g)
-    i_sort = sorted(i)
-    m_sort = sorted(m)
-    w_sort = sorted(w)
 
-    print("Writing out to json file")
+def to_json(g, i, m, w):
+    """
+    Dump state policy data to json file(s)
 
-    # one large json file
-    with open("data/abortion_policy_api.json", "w") as f:
-        json.dump(g_sort, f, indent=1)
-        json.dump(i_sort, f, indent=1)
-        json.dump(m_sort, f, indent=1)
-        json.dump(w_sort, f, indent=1)
+    Inputs:
+        g (dictionary): gestational limit policy by state
+        i (dictionary): insurance coverage policy by state
+        m (dictionary): minors policy by state
+        w (dictionary): waiting periods by state
+    """
 
-    # json file per policy area
+    print("Writing out to json file")
     with open("data/abortion_policy_api_gestational.json", "w") as f:
-        json.dump(g_sort, f, indent=1)
+        json.dump(g, f, indent=1)
 
     with open("data/abortion_policy_api_insurance.json", "w") as f:
-        json.dump(i_sort, f, indent=1)
+        json.dump(i, f, indent=1)
 
     with open("data/abortion_policy_api_minors.json", "w") as f:
-        json.dump(m_sort, f, indent=1)
+        json.dump(m, f, indent=1)
 
     with open("data/abortion_policy_api_waiting.json", "w") as f:
-        json.dump(w_sort, f, indent=1)
+        json.dump(w, f, indent=1)
 
 
-def fix_missing_data(g, i, m, w):
+def clean():
     """
-    Fill in missing characteristics for each state
-
-    Inputs:
-        g (list): gestational limit policy by state
-        i (list): insurance coverage policy by state
-        m (list): minors policy by state
-        w (list): waiting periods by state
+    Cleans API dataset by filling in missing values.
     """
 
-    # TODO
+    states = get_states("data/states.txt")
+
+    # get API data
+    policy_areas = []
+    with open("data/abortion_policy_api_gestational_unclean.json", "r") as f:
```

Comment:
Since there is no difference in the logic here, I think it'd make sense for the names of the files to be in a list.